balancer/roundrobin/roundrobin_test.go

```diff
@@ -473,3 +496,56 @@ func (s) TestAllServersDown(t *testing.T) {
 	}
 	t.Fatalf("Failfast RPCs didn't fail with Unavailable after all servers are stopped")
 }
+
+func (s) TestUpdateAddressAttributes(t *testing.T) {
+	r := manual.NewBuilderWithScheme("whatever")
+
+	test, err := startTestServers(1)
```

Comment:
Why use this instead of a stubserver?  I guess it already pushes metadata to a channel but I think a stubServer would still be better so it's clear from the code here what is happening.

Reply:
This is not in e2e test. There's no stubServer. Maybe we can move it to internal.

Reply:
Oh, I see.  But it *is* an end-to-end test - which is good, but the reason I was confused.